azerothcore/azerothcore-wotlk
# feat(Core/Creature): Implement quest_greetings table

* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/a64d1ec51eb11fd01a26d08a09de3d8d7dd89675)

* cherry-pick commit (https://github.com/TrinityCore/TrinityCore/commit/6dda09818d32fc715549cda3e8bd6fa806425b27)

Co-Authored-By: ForesterDev <11771800+ForesterDev@users.noreply.github.com>
Co-Authored-By: Giacomo Pozzoni <giacomopoz@gmail.com>



## Changes Proposed:
-  Implement quest_greetings table
-  https://github.com/azerothcore/wiki/pull/686

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Builds
- Tested in-game

![image](https://user-images.githubusercontent.com/24550914/152534778-8bd2960a-e285-4a82-89be-29e8abc7b48c.png)


## How to Test the Changes:


1. 
```sql
INSERT INTO `quest_greeting` (`ID`, `Type`, `GreetEmoteType`, `GreetEmoteDelay`, `Greeting`, `VerifiedBuild`) VALUES (341, 0, 0, 0, 'I don\'t have much time for idle talk, $N.  I\'ve got to get this bridge rebuilt before the rains come.  I\'ve finished every project on-time and under budget and I\'m not about to start slipping now.', 0);
```

2. .go c id 341
3. Open gossip and see his quest_greeting is showing if there are 2 or more options there

to get it to show:

.q add 345

Also check that other gossips and quest givers works as they should


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
